src/client/features/view/components/tooltips/formatArray.js

```diff
@@ -0,0 +1,21 @@
+const _ = require('lodash');
+
+//Push a metadata field to the bottom of the list
+function pushToBottom(data, field) {
+  let withoutField = _.filter(data, pair => pair[0] !== field);
+  let withField = _.filter(data, pair => pair[0] === field);
+  return _.concat(withoutField, withField);
+}
+
+//Push a collection of elements to the bottom of the list
+//Takes in a list of id's and pushes all matching items to the bottom of the list
+//Requires a valid list of ids
+function collectionToBottom(data, idList) {
+  idList.forEach(field => data = pushToBottom(data, field));
```

Comment:
Sorry I should have been more clear myself, but what I particularly don't understand is the mutation of data in that forEach line.

Reply:
forEach simply just iterates over the IdList which just contains a collection of keys that need to be pushed to the bottom. 

Upon each foreach call, pushToBottom is called which returns a whole new ordered array. Since data contains the original unordered array, data is modified to ensure for the next call to pushToBottom has the correct partially sorted list.

Reply:
So is something like this the same?

```js
const collectionToBottom = (data, criteriaList) => {

   const matched = data.filter(pair => criteriaList.includes(pair[0]));
   const notMatched = _.difference(data, matched);
   
   return notMatched.concat(matched);
};
```

Reply:
@d2fong That snippet works for pushing elements to the, but does not maintain order. The desired behaviour is that elements are pushed down in the order they appear in the criterial list.  Order is important is the list order defines what appears first in the tooltip. 

For example if the criteriaList was [obj2, obj1] and the data was [obj1, obj2, obj3], the desired result would be [obj3, obj2, obj1], but the given snippet would return [obj3, obj1, obj2] .

Reply:
@d2fong  Your method works with the desired result if 2 additional lines were to be added
```js 
function collectionToBottom(data, criteriaList) {
  let matched = data.filter(pair => criteriaList.includes(pair[0]));
  let notMatched = _.difference(data, matched);
  let matchedGrouped =  _.groupBy(matched, 0);
  let matchedSorted = _.map(criteriaList, index => matchedGrouped[index].shift());
  return notMatched.concat(matchedSorted);
}
```
Although, I'm not sure how much cleaner that is compared to the existing.

Reply:
What about this:
```js
const collectionToBottom = (data, criteriaList) => {

   const matched = data.filter(pair => criteriaList.includes(pair[0]));
   const notMatched = _.difference(data, matched);
   
   return notMatched.concat(_.sortBy(matched, pair => pair[0]);
};
```

Reply:
But I guess you want a specific order still even if they are grouped by criteria list.

e.g if you want C, A, B.  You want all C items to come first, and then all A items, etc,  then the above still doesnt work?

Reply:
Wouldn't that sort the matched list in lexicographic order? The desired order may not always alphabetical, as criterialList can be set to anything.

Reply:
```js
const collectionToBottom = (data, criteriaList) => {

   const matched = data.filter(pair => criteriaList.includes(pair[0]));
   const notMatched = _.difference(data, matched);
   
   return notMatched.concat(_.sortBy(matched, pair => criteriaList.indexOf(pair[0]));
};
```

Reply:
But do you use the component function pushToBottom in other places too?

Reply:
Didn't know you could user _.sortBy like that, but yeah that works and is a lot shorter/simpler that the 4 line solution I had. 

I have updated the function to follow the same pattern/style.

Reply:
@d2fong pushToBottom was only used for collectionToBottom. The current updated version of this PR no longer has pushToBottom.